### ISSUE TITLE
Adding a field as required when data exist crash

### ISSUE THREAD

**Issue by jph98** [strapi/strapi#6292]
**Describe the bug**

**Steps to reproduce the behavior**

1. Create a content-type A and a content type B with two fields each (any type).  Save both of these.  
2. Create a relation between Content Type A and B (One to Many)
3. Populate Content Type A with some data
4. Create a new **required** text field in content type B and save
5. Saving will not complete and result in an error (see below)
6. Platform is completely unusable - totally borked - no way to recover it

**Expected behavior**

I expected the new field to be added to the content type without error.

**Screenshots**

**Code snippets**

Log output here:

```
[2020-05-20T10:47:59.333Z] debug ⛔️ Server wasn't able to start properly.
[2020-05-20T10:47:59.334Z] error error: column "level" contains null values
    at Parser.parseErrorMessage (/srv/strapi/strapi-development/node_modules/pg-protocol/dist/parser.js:241:15)
    at Parser.handlePacket (/srv/strapi/strapi-development/node_modules/pg-protocol/dist/parser.js:89:29)
    at Parser.parse (/srv/strapi/strapi-development/node_modules/pg-protocol/dist/parser.js:41:38)
    at Socket.<anonymous> (/srv/strapi/strapi-development/node_modules/pg-protocol/dist/index.js:8:42)
    at Socket.emit (events.js:310:20)
    at addChunk (_stream_readable.js:286:12)
    at readableAddChunk (_stream_readable.js:268:9)
    at Socket.Readable.push (_stream_readable.js:209:10)
    at TCP.onStreamRead (internal/stream_base_commons.js:186:23)
> strapi-development@0.1.0 develop /srv/strapi/strapi-development
> strapi develop
```

**System**

- Node.js version: 12.16.2
- NPM version:
- Strapi version: 3.0.0-beta.20.3
- Database: postgres psql (PostgreSQL) 10.12 (Ubuntu 10.12-0ubuntu0.18.04.1)
- Operating system: Created via the Digital Ocean droplet

**Additional context**

**Comment by iraguhalionel** [comment]
@jph98 can you share more about what you are trying to accomplish?

**Comment by Brad-Christie** [comment]
Also curious, when you specified the field as required did you also provide a default value?

**Comment by jph98** [comment]
There wasn't a default value specified.  I can't reproduce this reliably unfortunately.  There are some very random conditions that cause this to crash horribly requiring manual intervention (I modified the generated model in order to get Strapi back on it's feet).

**Comment by derrickmehaffy** [comment]
Generally this only happens if you try to modify an existing model that already has data and set a field as required, in which case PG is throwing the error as you can't set `NOT NULL` on a column that has `NULL` values.

Either you need to set a default or adjust the existing values to contain something other than null. `Required` in the context between Strapi and SQL is simply `NOT NULL` restriction on the column.

**Comment by jph98** [comment]
Maybe warn the user if they're going to cause inconsistency via validiation rules?  I mean if the entire change is likely to bring down the platform it's good to be defensive in that nature.

You can't adjust the values after the fact, because the user interface is totally inaccessible.

**Comment by leccles1** [comment]
Same happened to me

Existing content type of 'article'

Wanted to switch to a slug-style url schema instead of article id.

So I created a new field, 'slug' of type UID. Dependent on another field 'title'.

No default value entered (Can't, I assume because the value has to be unique) but made it a required field.

Crashed Strapi admin interface, and until I cleaned up the model file, removing the newly added field. I was unable to start Strapi due to Postgres error.

```
ERROR: column "slug" contains null values

2020-09-10 08:50:49.559 UTC [34] STATEMENT: alter table "articles" add column "slug" varchar(255) not null
```
